Modules/_testmultiphase.c

```diff
@@ -969,3 +969,43 @@ PyInit__test_shared_gil_only(void)
 {
     return PyModuleDef_Init(&shared_gil_only_def);
 }
+
+
+#include "datetime.h"
+
+static PyObject *
+datetime_capi_pydate_check(PyObject *self, PyObject *op)
+{
+    PyDateTime_IMPORT;
+#ifdef Py_GIL_DISABLED
+    if (!PyDateTimeAPI && PyInterpreterState_Get() != PyInterpreterState_Main()) {
+        // Subinterpreter cannot import lagacy _datetime module and capsule.
+        PyErr_WriteUnraisable(NULL);
+        Py_RETURN_NONE;
+    }
+#endif
+    if (PyDate_Check(op)) {
+        Py_RETURN_NONE;
+    }
+    PyErr_SetString(PyExc_TypeError, "object is not instance of capi type");
+    return NULL;
+}
+
+static PyMethodDef datetime_capi_client_methods[] = {
+    {"pydate_check", datetime_capi_pydate_check, METH_O, NULL},
+    {0},
+};
+
+static PyModuleDef_Slot datetime_capi_client_slots[] = {
+    {Py_mod_multiple_interpreters, Py_MOD_PER_INTERPRETER_GIL_SUPPORTED},
+    {0, NULL},
+};
+
+static PyModuleDef datetime_capi_client_def = TEST_MODULE_DEF(
+    "_testmultiphase_datetime_capi_client", datetime_capi_client_slots, datetime_capi_client_methods);
+
+PyMODINIT_FUNC
+PyInit__test_datetime_capi_client(void)
+{
+    return PyModuleDef_Init(&datetime_capi_client_def);
+}
```

Comment:
_testmultiphase is meant to exercise the capabilities of the import machinery, so it seems like an odd place for this.  Why not add the method to Modules/_testcapi/datetime.c?